Resources/ts/router.test-d.ts

```diff
@@ -0,0 +1,92 @@
+import { expectType } from 'tsd';
+import { RoutesMap } from '../js/router';
+import { Route, Router, Routing } from './router';
+import routes from './routes.json';
+
+expectType<Router>(Router.getInstance());
+expectType<Router>(Routing);
+
+expectType<RoutesMap>(Routing.getRoutes());
+expectType<Route>(Routing.getRoute('homepage'));
+
+expectType<string>(Routing.getBaseUrl());
+Routing.setBaseUrl('');
+
+expectType<string>(Routing.getScheme());
+Routing.setScheme('https');
+
+expectType<string>(Routing.getHost());
+Routing.setHost('localhost');
+
+expectType<string | null>(Routing.getPort());
+Routing.setPort('1234');
+
+expectType<string | null>(Routing.getLocale());
+Routing.setLocale('en');
+
+Routing.setRoutingData(routes);
+Routing.setRoutingData({
+  base_url: '',
+  routes: {
+    homepage: { tokens: [['text', '/']], defaults: [], requirements: [], hosttokens: [], methods: [], schemes: [], },
+    admin_index: { tokens: [['text', '/admin']], defaults: [], requirements: [], hosttokens: [], methods: [], schemes: [], },
+    admin_pages: { tokens: [['text', '/admin/path']], defaults: [], requirements: [], hosttokens: [], methods: [], schemes: [], },
+    blog_index: { tokens: [['text', '/blog']], defaults: [], requirements: [], hosttokens: [['text', 'localhost']], methods: [], schemes: [], },
+    blog_post: {
+      tokens: [
+        ['variable', '/', '[^/]++', 'slug'],
+        ['text', '/blog'],
+      ],
+      defaults: [],
+      requirements: [],
+      hosttokens: [['text', 'localhost']],
+      methods: [],
+      schemes: [],
+    },
+    users_delete: {
+      tokens: [
+        ['text', '/delete'],
+        ['variable', '/', '[^/]++', 'id', true],
+        ['text', '/users']
+      ],
+      defaults: [],
+      requirements: [],
+      hosttokens: [],
+      methods: [
+        'DELETE'
+      ],
+      schemes: []
+    },
+    bazinga_jstranslation_js: {
```

Comment:
This route is not used in the test, and it is quite specific. What do you want to test with this route? I'd prefer it if you either remove it or replace it by a generic (not mentioning another library) route.

Reply:
I wanted to test a "real route", fortunately this route:
- uses string and boolean for `tokens`
- uses object for `defaults` and `requirements`

But ofc I can make this route generic